### ISSUE TITLE
Mocking breaks when `isolate: false, singleThread: true`, v1.1.2

### ISSUE THREAD

**Issue by AriPerkkio** [vitest-dev/vitest#4894]
Note: This issue is specifically about `isolate: false, singleThread: true` configuration. 

### Describe the bug

Likely regression from https://github.com/vitest-dev/vitest/pull/4664, cc. @Dunqing 

If two files are loading same file and the second one mocks that file, Vitest will throw `Error: [vitest] Cannot mock "../src/basic.js" because it is already loaded. Did you import it in a setup file?` error. Reproduction below has no setup files - just two test files importing same source file while second one mocks that file. 

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-scpuyh?file=test%2Fsecond.test.ts,test%2Ffirst.test.ts

### System Info

```shell
Stackblitz
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by AriPerkkio** [comment]
Coverage tests are failing randomly, depending which file Vitest's cache decides to run first:

https://github.com/vitest-dev/vitest/blob/a73c1c2ca3a3ca93b317657b9328889434c36344/test/coverage-test/test/no-esbuild-transform.test.js#L1-L10

**Comment by sheremet-va** [comment]
Not sure if it is a bug. If the file structure was a little bit different, there would've been a situation when mocking wouldn't have worked in Vitest 1.0 (before the error that catches such bugs/features was introduced): 

https://stackblitz.com/edit/vitest-dev-vitest-gxiruo?file=src%2Fbasic2.ts,package.json,test%2Ffirst.test.ts,test%2Fsecond.test.ts

**Comment by AriPerkkio** [comment]
Right, if this is expected behaviour then maybe the error message should be different when it's caused by disabled isolation. Seeing error message that mentions setupFiles is a bit misleading in cases like the minimal repro.

**Comment by sheremet-va** [comment]
> Right, if this is expected behaviour then maybe the error message should be different when it's caused by disabled isolation. Seeing error message that mentions setupFiles is a bit misleading in cases like the minimal repro.

It is not possible to know why the error happens.

**Comment by AriPerkkio** [comment]
The possible causes could still be limited based on configurations. If there are no setupFiles, do not mention that. If isolation is disabled, maybe add that one there.

**Comment by sheremet-va** [comment]
This should be fixed by #4905 

This error is now thrown only for isolated environments.

**Comment by Tohid001** [comment]
@sheremet-va  hey mate, I have written hundreds of test cases mocking react's particular api's that's needed mocking. And now all those tests are getting failed with this error thrown:
`Please, remove the import if you want static imports to be mocked, or clear module cache by calling "vi.resetModules()" before mocking if you are going to import the file again. See: https://vitest.dev/guide/common-errors.html#cannot-mock-mocked-file-js-because-it-is-already-loaded`

I have tried the resetmodules solution but it did not work for me

**Comment by divmgl** [comment]
We're having the same problem. In the v0 version of Vitest things work as expected. The latest version of Vite fails with this same error.

**Comment by i-void** [comment]
I have the same issue. Reproducing is easy. In my case creating a test file with a mock for top level imported module and disable isolation (singleFork or isolation: false) is enough to see the error. For just maxFork: 1 minFork. 1 it works but this time tests are 10x slower.

**Comment by divmgl** [comment]
> Please, provide reproductions when filing an issue. Any comment that basically means "+1" will be _ignored_.

I went into StackBlitz and attempted to recreate the issue to no avail. I spent some time in our codebase commenting out code until I narrowed down where it was happening. We have a complex web of dependencies and I noticed one function that was exported in a module that had a ton of different imports. I moved that function into a different module and the issue stopped happening.

If I had to guess, this issue occurs when there's a circular dependency between two modules that indirectly import the mocked module. Hopefully this helps as I won't be able to spend any more time on this. cc @sheremet-va
